libs/design-system/src/pagination/usePaginationState.ts

```diff
@@ -50,8 +56,14 @@ function getValidatedPageNumberFromUrl(pageNumberStr: string | undefined, starti
   return pageNumberUnchecked && pageNumberUnchecked > 0 ? pageNumberUnchecked : startingPageNumber;
 }
 
-function getValidatedPageSizeFromUrl(pageSizeStr: string | undefined, pageSizes: number[]): number {
+function getValidatedPageSizeFromUrl(
+  pageSizeStr: string | undefined,
+  startingPageSize: number,
+  pageSizes: number[]
+): number {
   const sizeValUnchecked = parseInt(pageSizeStr ?? 'NaN', 10);
 
-  return pageSizes.includes(sizeValUnchecked) && sizeValUnchecked > 0 ? sizeValUnchecked : pageSizes[0];
+  return pageSizes.includes(sizeValUnchecked) && sizeValUnchecked > 0
+    ? sizeValUnchecked
+    : startingPageSize ?? pageSizes[0];
```

Comment:
use the `startingPageSize` if search params don't contain the value or default page size value